### PR TITLE
fixing duplicates and code smell

### PR DESCRIPTION
**Pull request by bishoymelek**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -6,12 +6,12 @@ module.exports = Redis;
...
   global.R5 = {
     out: console
-  }
+  };
 }
 
 // Constructors
 
-function Redis (host, port, pass, db = 0) {
+function Redis(host, port, pass, db = 0) {
```
Comment (RaviPatel) [inline index.js]: minor: i prefer the spaces here

File: index.js [hunk 2]
```diff
@@ -23,8 +23,20 @@ function Redis (host, port, pass, db = 0) {
 // Public Methods
 // TODO: check if this.ready, else reconnect and/or queue?
 
-Redis.prototype.connect = function () {
-  if (this.ready) { return; }
+function doesCallbackExist(callback) {
```
Comment (RaviPatel) [inline index.js]: minor: I prefer snake_case instead of camelCase

File: index.js (hunk 1 above)
Comment (bishoymelek) [reply index.js]: fixed

File: index.js [hunk 3]
```diff
@@ -1,17 +1,18 @@
-/* eslint-disable brace-style, camelcase, semi */
+/* eslint-disable no-use-before-define */
```
Comment (RaviPatel) [inline index.js]: let's merge these two lines:
```
eslint-disable brace-style, camelcase, no-use-before-define, semi
```

File: index.js [hunk 4]
```diff
@@ -1,17 +1,18 @@
...
+/* eslint-disable  brace-style, camelcase, semi */
 /* global R5 */
 
 module.exports = Redis;
 
 if (!global.R5) {
   global.R5 = {
-    out: console
-  }
+    out: console,
```
Comment (RaviPatel) [inline index.js]: we shouldn't have a comma here, right?

File: index.js [hunk 5]
```diff
@@ -23,85 +24,97 @@ function Redis (host, port, pass, db = 0) {
...
     }
-  }
-  else {
+  } else {
     this.client.set(key, value, handle_data);
   }
 
-  function handle_data (err, data) {
-    if (err) { R5.out.error(err); }
+  function handle_data(err, data) {
```
Comment (RaviPatel) [inline index.js]: we need to have all spaces between all of these.. basically, the rule is to have a space when defining the function and no space when calling the function. E.g., 
```
function handle_data (err, data) { ... }
...
handle_data('an-error', 'some-data');
```

File: index.js [hunk 6]
```diff
@@ -1,12 +1,12 @@
-/* eslint-disable brace-style, camelcase, semi */
+/* eslint-disable  brace-style, camelcase, semi, no-use-before-define */
```
Comment (RaviPatel) [inline index.js]: let's actually remove `no-use-before-define` as I don't think we need it

Comment (bishoymelek) [reply index.js]: I used as I was using Airbnb code styling so it was needed, now removed it as we use standardjs

File: index.js [hunk 7]
```diff
@@ -33,75 +45,75 @@ Redis.prototype.connect = function () {
...
 
-  let _this = this;
+  const _this = this;
   this.client.on('ready', function () {
     R5.out.log(`Connected to Redis (db: ${this.db})`);
     _this.ready = true;
   });
 
-  this.client.on('error', function (err) {
+  this.client.on('error', err => {
```
[the remark this answers is not in the document]
Comment (bishoymelek) [reply index.js]: fixed

File: test/crudList.js [hunk 8]
```diff
@@ -0,0 +1,114 @@
+/* eslint-disable brace-style, camelcase, semi */
+/* eslint-env mocha */
+
```
[the remark this answers is not in the document]
Comment (bishoymelek) [reply test/crudList.js]: fixed

File: index.js [hunk 9]
```diff
@@ -33,75 +35,75 @@ Redis.prototype.connect = function () {
...
+    );
     value = stringify(value);
   }
 
   if (key_expiration) {
     if (value) {
       this.client.set(key, value, 'EX', key_expiration, handle_data);
-    }
-    else {
+    } else {
```
Comment (RaviPatel) [inline index.js]: I prefer ifs/elsifs/elses on separate lines from the brackets:
```
if () {
  ...
}
else if () {
  ...
}
else {
  ...
}
```

Comment (bishoymelek) [reply index.js]: So that we should be using brace-style stroustrup, Correct?

**Pull request merged**